components/claim/components/email_to_oauth.tsx

```diff
@@ -12,26 +11,34 @@ import {t} from 'utils/i18n.jsx';
 import LoginMfa from 'components/login/login_mfa.jsx';
 import LocalizedInput from 'components/localized_input/localized_input';
 
-export default class EmailToOAuth extends React.PureComponent {
-    static propTypes = {
-        newType: PropTypes.string,
-        email: PropTypes.string,
-        siteName: PropTypes.string,
-    };
+export type State = {
+    showMfa?: boolean;
+    password?: string;
+    error?: string;
+};
 
-    constructor(props) {
-        super(props);
+type Props = {
+    email: string;
+    newType: string;
+    siteName: string;
+}
+export default class EmailToOAuth extends React.PureComponent <Props, State> {
+    private passwordInput: React.RefObject<HTMLInputElement>;
 
-        this.state = {showMfa: false, password: ''};
```

Comment:
Maybe I'm wrong, but don't we need to set `this.state` in the constructor? Maybe something changed with React in a newer version, but I thought we had to set it to at least an empty object if we're using it in a component.

Reply:
Ok my mistake for this I didn't set state object. But making state object give me this kinds of error. How do I resolve it?
<img width="345" alt="Screen Shot 2021-07-26 at 11 14 12" src="https://user-images.githubusercontent.com/35423212/126932322-7f2b0e6b-cd92-45cc-99bc-3d8d3e5a14fd.png">
<img width="473" alt="Screen Shot 2021-07-26 at 11 13 58" src="https://user-images.githubusercontent.com/35423212/126932326-a813274b-d3f7-41f2-bb70-439e5144ab3a.png">
<img width="331" alt="Screen Shot 2021-07-26 at 11 13 46" src="https://user-images.githubusercontent.com/35423212/126932327-a8365f2b-8329-43a3-995b-0ea55759bdf2.png">
<img width="388" alt="Screen Shot 2021-07-26 at 11 13 36" src="https://user-images.githubusercontent.com/35423212/126932329-0876bd20-9673-4b0e-9c5b-ceebf16ca341.png">

Reply:
without seeing the new code you are adding, I'd say you probably need to call `super(props)` first thing in the constructor